developer-hub/src/reference_docs/extrinsic_encoding.rs

```diff
@@ -0,0 +1,156 @@
+//! # Constructing and Signing Extrinsics
+//!
+//! Substrate is configurable enough that extrinsics can take any format, in theory. In practice,
+//! runtimes tend to use our [`sp_runtime::generic::UncheckedExtrinsic`] type to represent extrinsics.
+//! In Polkadot, this is configured
+//! [here](https://github.com/polkadot-fellows/runtimes/blob/94b2798b69ba6779764e20a50f056e48db78ebef/relay/polkadot/src/lib.rs#L1478)
+//! at the time of writing.
+//!
+//! What follows is a description of how extrinsics based on this type are encoded into bytes. These
+//! bytes can then be submitted to a chain for inclusion in a block (this is how we make changes to
+//! the state of a chain), and are ultimately stored in the body of a block.
+//!
+//! # Encoding an Extrinsic
+//!
+//! At a high level, all extrinsics are formed from concatenating some details together, ie:
+//!
+//! ```text
+//! extrinsic_bytes = concat(
```

Comment:
I think this is not arbitrary though, I would show an actual `Encode` impl block here.

Reply:
I didn't do this initially because I felt the the Encode impl is a little over-complicated unless you understand what's going on, ie it's:

```rust
fn encode(&self) -> Vec<u8> {
	let mut tmp = Vec::with_capacity(sp_std::mem::size_of::<Self>());

	// 1 byte version id.
	match self.signature.as_ref() {
		Some(s) => {
			tmp.push(EXTRINSIC_FORMAT_VERSION | 0b1000_0000);
			s.encode_to(&mut tmp);
		},
		None => {
			tmp.push(EXTRINSIC_FORMAT_VERSION & 0b0111_1111);
		},
	}
	self.function.encode_to(&mut tmp);

	let compact_len = codec::Compact::<u32>(tmp.len() as u32);

	// Allocate the output buffer with the correct length
	let mut output = Vec::with_capacity(compact_len.size_hint() + tmp.len());

	compact_len.encode_to(&mut output);
	output.extend(tmp);

	output
}
```

Perhaps it's worth including that snippet of code here, but if so, I'd be tempted to have it in addition to the pseudocode, which I also use to name the parts of each of the sections below. Alternately I could make this pseudocode more rusty!

Reply:
I kinda agree that the full impl is too long, and leave it up you to decide. 

an impl block can indeed be referenced in docify, see an example here: https://docs.rs/docify/latest/docify/attr.export.html

Reply:
For now, I added it!